### PR TITLE
Completed express-notes-api. | Vera Yeh

### PR DESCRIPTION
**Pull request by VeraYeh**
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image
image

### CODE REVIEW

File: express-notes-api/index.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+})
+
+// 1) Clients can GET a list of notes.
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (prop in data.notes) {
+    notesArray.push(data.notes[prop]);
+  }
+  res.status(200).json(notesArray);
+})
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:18]: "Clients can GET a list of notes" looks good.

File: express-notes-api/index.js [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+// 2) Clients can GET a single note by id.
+app.get('/api/notes/:id', (req, res) => {
+  const requestID = req.params.id;
+
+  // if requested id not a positive integer
+  if (isNaN(requestID) || Number(requestID) <= 0) {
+    noteObj = { error: 'id must be a positive integer' };
+    res.status(400).json(noteObj);
+    return;
+  };
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:29]: What about if `requestID` is a positive non-integer number (ex: `4.23`)?

File: express-notes-api/index.js [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+    // If client includes a content property in the request body and the note is not successfully recorded, 500 response
+      if (err) {
+        noteObj = { error: 'An unexpected error occurred' };
+        res.status(500).json(noteObj);
+      } else {
+        res.status(201).json(noteObj);
+      }
+    })
+  }
+})
```
Comment (TimelikeClosure) [inline express-notes-api/index.js:69]: "Clients can POST a new note" looks good.

**Review by TimelikeClosure — CHANGES_REQUESTED** [review]

File: express-notes-api/index.js (hunk 2 above)
Comment (VeraYeh) [reply express-notes-api/index.js]: @TimelikeClosure
Update line26 with additional condition to check if it's an integer number.
image

**Review by TimelikeClosure — APPROVED** [review]

**Pull request merged**